classes/Infrastructure/Auth/CsrfCheck.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace OpenCFP\Infrastructure\Auth;
+
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\Security\Csrf\CsrfToken;
+use Symfony\Component\Security\Csrf\CsrfTokenManager;
+
+class CsrfCheck
+{
+    /**
+     * @var CsrfTokenManager
+     */
+    private $manager;
+
+    public function __construct(CsrfTokenManager $manager)
+    {
+        $this->manager = $manager;
+    }
+
+    public function checkCsrf($tokenId, $token)
+    {
+        $token = new CsrfToken($tokenId, $token);
+        if (!$this->manager->isTokenValid($token)) {
+            return new RedirectResponse('/dashboard');
```

Comment:
How about turning this entire class into a validator that has an `isValid()` method, which accepts a `Request` object, returns a `bool`?

The advantage would be that this class would have a single responsibility, which is to validate a request object, and not need to return a `RedirectResponse` here.